### ISSUE TITLE
Test error

### ISSUE THREAD

**Issue by reid3290** [yanshao9798/tagger#4]
Hi Yan, I successfully trained the model with my own corpus, but when I tested it, I encountered an error like following:
![image](https://user-images.githubusercontent.com/9301137/33644078-bb6509f8-da7d-11e7-9928-b82d716a4f46.png)
My test command is  
```python tagger.py test -p PFR -e test.txt -m PFR -emb Embeddings/glove.txt```
Do you have any idea about this?
Thank you.

**Comment by yanshao9798** [comment]
It looks like the tag 's]ns' only appears in the testing data while not found in the training data. This tag also looks really weird to me. May I ask what data set are you applying?

**Comment by reid3290** [comment]
Hi Yan, Thanks for your reply.
I'm using PFR corpus(http://www.icl.pku.edu.cn/icl_res/). The link is invalid and I downloaded the corpus from other's baidu driver.
I checked the corpus, there are some wired annotions:
```
[陕甘宁_j 边区_s]ns
```
I manually changed it to
```
陕甘宁_j 边区_s
```
The previous error is gone but I got a new one:
![image](https://user-images.githubusercontent.com/9301137/33672288-d9cf1674-dae4-11e7-91d0-358653771f56.png)

Apparently, it's something about the tag `Vg`. I checked it too, but it does appears in the training data. Can you think of any other possible cause?
`

**Comment by yanshao9798** [comment]
[陕甘宁_j 边区_s]ns seems to be named entity annotation. If you don't care about named entity recognition, it is safe to discard the relevant information.

For the tag Vg, you can check if all the associated words in the training corpus are single character-words. If so, the combinatory tags are pruned (You can find the details in Section 2.2 of the paper). This will cause problems if there are two or more character Vg words in the test data. This is my best guess. 

There are two ways to solve the problem:
1. Modify get_comb_tags() in toolbox.py so that it doesn't prune the "redundant" tags and retrain the model. But I wouldn't recommend so because it slows down the training speed and may lead to performance loss.
2. Input the raw test data and tag it with tag (instead of test) option and then evaluate the output afterwards.

Hope it helps!

**Comment by reid3290** [comment]
I removed all the named entity annotations in my corpus and then it works perfectly. 
Thank you.
